e2e/scripts/setup-local-fdb-cluster.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+# This source file is part of the FoundationDB open source project
+#
+# Copyright 2023 Apple Inc. and the FoundationDB project authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script works out of the box for Linux environments (tested on Ubuntu 16.04.5 LTS), after installing dependent software (kind, Docker, and kustomize);
+# Running in other environments may involve non-trivial debugging effort.
+
+# Set up local FDB non-HA cluster on a 4-node kind k8s cluster
+# This only works on x86 machines as FDB doesn't provide arm64 Linux binaries yet.
+set -eo errexit
+
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
+cd "${SCRIPT_DIR}"
+
+# Kubernetes version for the Kind clusters
+KUBE_VERSION=${KUBE_VERSION:-"v1.24.7"}
+# Kind cluster name
+CLUSTER=${CLUSTER:-"local-cluster"}
+
+if ! kind get clusters | grep "${CLUSTER}" &> /dev/null; then
+    echo "===Start creating k8s cluster on kind"
+    cat <<EOF | kind create cluster --name "${CLUSTER}" --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  image: kindest/node:${KUBE_VERSION}
+- role: worker
+  image: kindest/node:${KUBE_VERSION}
+- role: worker
+  image: kindest/node:${KUBE_VERSION}
+- role: worker
+  image: kindest/node:${KUBE_VERSION}
+EOF
+else
+    echo "===Kind cluster already exists"
+fi
+
+kubectl config use-context "kind-${CLUSTER}"
+
+echo "===Start building operator"
```

Comment:
Once I'm back I'm going to remove some parts of this script here and move it to a document. This currently breaks some stuff in our GitHub actions (and was flaky before that).

Reply:
which part is flaky?

Reply:
I left the script as it is but removed the usage from the CI. The setup had the following problems:

1.) We installed the latest CRD directly, so this check: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/.github/workflows/pull_request.yml#L87-L101 was broken.
2.) Based on all runs that I have seen the FDB cluster was never running: https://github.com/FoundationDB/fdb-kubernetes-operator/actions/runs/4669101303/jobs/8267089024#step:7:2729, creating the cluster without any further checks brings no value.